packages/app/ui/buttons/walletconnectBtn.tsx

```diff
@@ -1,47 +1,44 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import WalletConnectIcon from "app/ui/icons/walletconnect-icon";
-// import QRCodeModal from '@walletconnect/qrcode-modal'
-// import { signManageDataOp } from "app/stellar"
-// import { WalletConnectStore } from 'app/state/wallet-connect'
-import { Pressable, Text } from "app/design-system";
+import { Button } from "app/design-system";
+import { useWalletConnectClient } from "app/provider/WalletConnect";
 
-const WalletConnectBtn = ({}: { signInWithXDR?: (_: any) => {} }) => {
-  // const state = "idle"
-  // const { uri, signXdr, publicKey, connect, state } = WalletConnectStore()
-
-  // useEffect(() => {
-  //   if (!uri) return QRCodeModal.close()
-  //   QRCodeModal.open(uri, () => {}, {
-  //     desktopLinks: [],
-  //     mobileLinks: ['lobstr'],
-  //   })
-  // }, [uri])
+type Props = {
+  onConnected: (_publicKey: string) => void;
+  disabled?: boolean;
+  loading?: boolean;
+};
 
-  // const handleSignInWithXdr = async (publicKey: string) => {
-  //   const xdr = await signManageDataOp(publicKey)
-  // const { signedXDR } = await (signXdr(xdr) as Promise<{ signedXDR: string }>)
-  // signInWithXDR && signInWithXDR(signedXDR)
-  // }
+export const WalletConnectBtn = ({ onConnected, disabled, loading }: Props) => {
+  const [waitingForApproval, setWaitingForApproval] = useState<boolean>(false);
+  const { connect, accounts } = useWalletConnectClient();
 
-  const handleConnect = () => {};
+  useEffect(() => {
+    if (accounts.length && waitingForApproval) {
+      setWaitingForApproval(false);
+      const publicKey = accounts[0]!.replace("stellar:pubnet:", "");
+      onConnected(publicKey);
+    }
+  }, [accounts, onConnected, waitingForApproval]);
 
-  // useEffect(() => {
-  //   if (publicKey && signInWithXDR) handleSignInWithXdr(publicKey)
-  // }, [publicKey])
+  const onPress = () => {
+    if (!accounts.length) {
+      setWaitingForApproval(true);
+      connect();
+    } else {
+      const publicKey = accounts[0]!.replace("stellar:pubnet:", "");
```

Comment:
stellar:pubnet it probably shouldn't be hardcoded here but read from config

Reply:
The only way to test right now with Lobstr wallet is with pubnet, that's probably why, I do see the variable in the config file.

Reply:
yes, basically what we want here is the publicKey. Wallets differ from each other. There are wallets that allow requesting connection to both chains at once. Lobstr supports only `pubnet`. xBull supports both `pubnet` and `testnet`, but allows only one at a time. It was actually the reason why xBull was not working for me before because I wanted to connect to both `testnet` and `pubnet`. I've fixed it and xBull should now be working with our app!